### ISSUE TITLE
🐛  Source Zendesk Support/Chat: got stuck after 100k records

### ISSUE THREAD

**Issue by marcosmarxm** [airbytehq/airbyte#12591]
Is this your first time deploying Airbyte: No
OS Version / Instance: Ubuntu 20.04
Memory / Disk: 8Gb / 100 GB ssd
Deployment: Docker
Airbyte Version: 0.35.61-alpha
Source name/version: airbyte/source-zendesk-chat:0.1.6 (latest)
Destination name/version: airbyte/destination-bigquery: 1.0.5 (latest)
Step: normal sync.
Description: everytime the process starts to sync it freezes after some hours, the same happens with source-zendesk-support:0.2.3 (latest).

See discussion in Discourse: https://discuss.airbyte.io/t/zendesk-chat-and-zendesk-support-freezes-after-some-hours/455

**Comment by RobertoBonnet** [comment]
Hello guys. I have a solution for Zendesk Chat

**Comment by RobertoBonnet** [comment]
TL DR;
- Zendesk API changed the rate limit for offset pagination. Rate limit down to 10 request/minute after page 1000 -https://support.zendesk.com/hc/en-us/articles/4408824785946-Announcing-API-rate-limits-for-Offset-based-Pagination
- Sleep to asynchronous requests with 8 works (default) is a problem -https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py#L275

I started to debug the connector to Zendesk suport. The problem only affect clients with to much data, indeed.


Some routes requests are mades using asynchronous requests. This work to small data, but the process is stucking everything when the error start for large data.

Searching I have found some changed in Zendesk API that affect this code. The rate limit change for pagination greater than 1000. Here at Hurb we are making some changes.
SatisfactionRatings - Process request by request using cursor-based pagination
TicketMetrics - The cursor-based pagination aren't working here. There is a error in Zendesk DB. To avoid the problem, we are processing request by request applying the rule of 10 request per minute after pagination 1000 using offset-based pagination

**Comment by RobertoBonnet** [comment]
There is a way to see the error. Just make a request to https://{account}.zendesk.com/api/v2/ticket_metrics.json?page=999999 11 times in a minute

**Comment by marcosmarxm** [comment]
Thanks @RobertoBonnet to dive into this! Really helpful analysis

**Comment by marcosmarxm** [comment]
> There is a way to see the error. Just make a request to https://{account}.zendesk.com/api/v2/ticket_metrics.json?page=999999 11 times in a minute

Should we add this new request limits to your PR?

**Comment by RobertoBonnet** [comment]
@marcosmarxm It won't be needed by the tests I performed. The way the data is fetched does not go through this rate limit. I made some changes and I'm processing the full sync. Waiting to see what will happen. 3 days have passed.

![image](https://user-images.githubusercontent.com/19189101/169833234-35e3d7fd-5531-41d7-888a-39c45401a40e.png)

**Comment by RobertoBonnet** [comment]
The task failed :(. Is there any strategy to not lose all the data as downloaded? There are so many data from Zendesk here and, when the task faills, we lost to many time. I'll try to limit the pagination of each router from api that we use incremental.

![image](https://user-images.githubuserc
[logs-428.txt](https://github.com/airbytehq/airbyte/files/8765404/logs-428.txt)
ontent.com/19189101/170107343-cf245520-f3d8-4530-84e0-e3e05c62a782.png)

**Comment by RobertoBonnet** [comment]
[Uploading logs-428.txt…]()

**Comment by marcosmarxm** [comment]
> The task failed :(. Is there any strategy to not lose all the data as downloaded? There are so many data from Zendesk here and, when the task faills, we lost to many time. I'll try to limit the pagination of each router from api that we use incremental.
> 
> ![image](https://user-images.githubusercontent.com/19189101/170107470-d4c2aed9-05c5-4592-9228-eae8af943f8b.png)

You must use checkpoint to send data to destination and next time dont need to start from scratch. but only works with incremental streams

**Comment by bazarnov** [comment]
> Uploading logs-428.txt…

@RobertoBonnet Please attach the valid logs, since i'm afraid this link leads to other PR - not logs)

**Comment by RobertoBonnet** [comment]
@bazarnov Sorry... I fixed the link

**Comment by bazarnov** [comment]
Similar issue: https://github.com/airbytehq/airbyte/issues/12155
